From 2b2ac8471fed70da6af1bd261c808b9106e54c37 Mon Sep 17 00:00:00 2001
Message-Id: <patch-1.1-2b2ac8471f-20210408T132357Z-avarab@gmail.com>
In-Reply-To: <patch-1.1-ea968affa8c-20210324T233254Z-avarab@gmail.com>
References: <patch-1.1-ea968affa8c-20210324T233254Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Wed, 24 Mar 2021 16:55:42 +0100
Subject: [PATCH v2] config.c: remove last remnant of GIT_TEST_GETTEXT_POISON
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Remove a use of GIT_TEST_GETTEXT_POISON added in f276e2a4694 (config:
improve error message for boolean config, 2021-02-11).

This was simultaneously in-flight with my d162b25f956 (tests: remove
support for GIT_TEST_GETTEXT_POISON, 2021-01-20) which removed the
rest of the GIT_TEST_GETTEXT_POISON code.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
Range-diff:
1:  ea968affa8 ! 1:  2b2ac8471f config.c: remove last remnant of GIT_TEST_GETTEXT_POISON
    @@ config.c: int git_config_bool(const char *name, const char *value)
      	return v;
      }
      
    -
    - ## t/t1300-config.sh ##
    -@@ t/t1300-config.sh: test_expect_success 'invalid unit boolean' '
    - 	git config commit.gpgsign "1true" &&
    - 	test_cmp_config 1true commit.gpgsign &&
    - 	test_must_fail git config --bool --get commit.gpgsign 2>actual &&
    --	test_i18ngrep "bad boolean config value .1true. for .commit.gpgsign." actual
    -+	grep "bad boolean config value .1true. for .commit.gpgsign." actual
    - '
    - 
    - test_expect_success 'line number is reported correctly' '

 config.c | 16 +---------------
 1 file changed, 1 insertion(+), 15 deletions(-)

diff --git a/config.c b/config.c
index 6428393a41..870d9534de 100644
--- a/config.c
+++ b/config.c
@@ -1180,20 +1180,6 @@ static void die_bad_number(const char *name, const char *value)
 	}
 }
 
-NORETURN
-static void die_bad_bool(const char *name, const char *value)
-{
-	if (!strcmp(name, "GIT_TEST_GETTEXT_POISON"))
-		/*
-		 * We explicitly *don't* use _() here since it would
-		 * cause an infinite loop with _() needing to call
-		 * use_gettext_poison().
-		 */
-		die("bad boolean config value '%s' for '%s'", value, name);
-	else
-		die(_("bad boolean config value '%s' for '%s'"), value, name);
-}
-
 int git_config_int(const char *name, const char *value)
 {
 	int ret;
@@ -1268,7 +1254,7 @@ int git_config_bool(const char *name, const char *value)
 {
 	int v = git_parse_maybe_bool(value);
 	if (v < 0)
-		die_bad_bool(name, value);
+		die(_("bad boolean config value '%s' for '%s'"), value, name);
 	return v;
 }
 
-- 
2.31.1.527.g9b8f7de2547

